bitnami/spark/templates/deployment-master.yaml

```diff
@@ -0,0 +1,169 @@
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/component: master
+spec:
+  replicas: 1
+  strategy:
+    type: {{ .Values.master.updateStrategy }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "spark.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+      app.kubernetes.io/component: master
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "spark.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        helm.sh/chart: {{ include "spark.chart" . }}
+        app.kubernetes.io/component: master
+    {{- if .Values.master.nodeSelector }}
+      nodeSelector:
+      {{ toYaml .Values.master.nodeSelector | nindent 8 }}
+    {{- end }}
+    {{- with .Values.master.affinity }}
+      affinity:
+      {{ toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .Values.master.tolerations }}
+      tolerations:
+      {{ toYaml . | nindent 8 }}
+    {{- end }}
+    spec:
+      {{- include "spark.imagePullSecrets" . | nindent 6 }}
+      {{- if .Values.master.securityContext.enabled }}
+      securityContext:
+        fsGroup: {{ .Values.master.securityContext.fsGroup }}
+        runAsUser: {{ .Values.master.securityContext.runAsUser }}
+      {{- end }}
+      containers:
+        - name: spark-master
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.master.webPort }}
+              protocol: TCP
+            - name: cluster
+              containerPort: {{ .Values.master.clusterPort }}
+          volumeMounts:
+          {{- if .Values.master.configurationConfigMap }}
+            - name: config
+              mountPath: '/opt/bitnami/spark/conf/'
+          {{- end }}
+          {{- if .Values.security.certificatesSecretName }}
+            - name: certs
+              mountPath: '/opt/bitnami/spark/conf/certs'
+              readOnly: true
+          {{- end }}
+          env:
+          - name: SPARK_MODE
+            value: "master"
+          {{- if .Values.master.debug }}
+          - name: BASH_DEBUG
+            value: "1"
+          - name: NAMI_DEBUG
+            value: "1"
+          - name: NAMI_LOG_LEVEL
+            value: "trace8"
+          {{- end }}
+          - name: SPARK_DAEMON_MEMORY
+            value: {{ .Values.master.daemonMemoryLimit | quote }}
+          {{- if .Values.master.clusterPort }}
+          - name: SPARK_MASTER_PORT
+            value: {{ .Values.master.clusterPort | quote }}
+          {{- end }}
+          {{- if .Values.master.webPort }}
+          - name: SPARK_MASTER_WEBUI_PORT
+            value: {{ .Values.master.webPort | quote }}
+          {{- end }}
+          - name: SPARK_MASTER_OPTS
+            value: {{ .Values.master.configOptions | quote }}
```

Comment:
by default, `.Values.master.configOptions` does not have any value so I would not set the environment variable